models/articles.models.js

```diff
@@ -14,32 +14,49 @@ exports.selectArticleById = (article_id) => {
     });
 };
 
-exports.selectArticles = (sort_by, order) => {
+exports.selectArticles = (sort_by, order, topic) => {
+  const queryProms = [];
+  const queryVals = [];
   let queryStr =
-    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url";
+    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id";
 
-  const validColumns = ["author", "title", "article_id", "topic", "created_at", "votes", "article_img_url"]
+  if (topic) {
+    queryProms.push(checkValueExists("articles", "topic", topic));
+    queryVals.push(topic);
+    queryStr +=
+      " WHERE topic = $1 GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url";
+  } else {
+    queryStr +=
+      " GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url";
+  }
```

Comment:
You always want the GROUP BY included, so I would just add the WHERE clause in the if statement and then without an else, always add the GROUP BY. Then you don't have to repeat yourself